### ISSUE TITLE
No semantic coloring in the (new in 15.3) "Blue (extra contrast)" theme

### ISSUE THREAD

**Issue by vasily-kirichenko** [Microsoft/visualfsharp#3436]
![image](https://user-images.githubusercontent.com/873919/29328442-3a328d90-81fb-11e7-99c7-a37392caaf1b.png)

**Comment by cartermp** [comment]
I can't repro this on 15.4.1.17081501, targeting net47.  New project, open existing, change theme back and forth all works.

![no-repro-colors](https://user-images.githubusercontent.com/6309070/29332626-88013b34-81b5-11e7-9b11-6814e3bf01b2.PNG)

**Comment by vasily-kirichenko** [comment]
![1](https://user-images.githubusercontent.com/873919/29332966-a4361f2c-820a-11e7-8503-c762ce140f81.gif)

**Comment by vasily-kirichenko** [comment]
Maybe the colors are too close to black? I'm not sure.

**Comment by cartermp** [comment]
Dunno.  Your first screenshot definitely had no semantic colorization though, so it seems that it wasn't triggered then.

**Comment by cartermp** [comment]
I'm able to see this [here](https://github.com/fjoppe/FsYamlParser#L27):

![repro-colors-blue-high-contrast](https://user-images.githubusercontent.com/6309070/29341785-8ab1e8ee-81db-11e7-9106-268eb90f78b2.PNG)

Semantic color for `TextReader` works fine on other themes.

**Comment by Pilchie** [comment]
Not sure how that could happen.  Tagging @dpoeschl for ideas.  Is `TextReader` normally faded in other themes for some reason?

**Comment by cartermp** [comment]
Nope.  Good elsewhere.

Dark:
![theme-dark](https://user-images.githubusercontent.com/6309070/29341924-6d38658a-81dc-11e7-8441-c8fa4613bb29.PNG)

Blue: 
![theme-blue](https://user-images.githubusercontent.com/6309070/29341928-71ecff5a-81dc-11e7-9bdc-52dc3da78bde.PNG)

White:
![theme-white](https://user-images.githubusercontent.com/6309070/29341931-75d9c828-81dc-11e7-9008-eb3a52df694a.PNG)

**Comment by vasily-kirichenko** [comment]
We have a distinct color category for disposable types, hence the bug. We need to add this new theme to our list of "light" themes to assign proper default values to our custom categories.

**Comment by Pilchie** [comment]
@dpoeschl - can you let @brettfo know what needs to happen here?

**Comment by dpoeschl** [comment]
@brettfo Can we just add this to EditorColors.pkgdef like we did with the yellow color earlier? Or are you trying to theme it In The Open?

**Comment by vasily-kirichenko** [comment]
We detect that a theme is "light" here https://github.com/Microsoft/visualfsharp/blob/master/vsintegration/src/FSharp.Editor/Classification/ClassificationDefinitions.fs#L68

The new "Blue (high contrast)" one has the following id:

![image](https://user-images.githubusercontent.com/873919/29373136-e9a935ac-82b5-11e7-8946-89628395aa8b.png)

`{ce94d289-8481-498b-8ca9-9b6191a315b9}`

I added handling it in https://github.com/Microsoft/visualfsharp/pull/3443, but it's not helped so far.

**Comment by vasily-kirichenko** [comment]
`KnownColorThemes` does not contain this Guid (`HighContrast` is a different one):

![image](https://user-images.githubusercontent.com/873919/29375864-3992618e-82bf-11e7-8fd0-4adacc05b20d.png)

**Comment by dpoeschl** [comment]
Right, it doesn't look like KnownColorThemes was updated to include the new theme in 15.3.

> We need to add this new theme to our list of "light" themes to assign proper default values to our custom categories.

Except the "Blue (Extra Contrast)" theme isn't just any old light theme. It's kind of a light theme but one that enforces a 4.5:1 contrast ratio for *everything*. We had to make the mutible var color #785500 instead of #A08000 in the Blue (Extra Contrast) theme for this, which we accomplish in closed code via EditorColors.pkgdef.

To do this in your open ClassificationDefinitions setup, you'd first have to detect the theme (using the magic guid since KnownColorThemes isn't updated), but then not just consider it a regular old "Light theme" but call it something different, and then fix up the non-4.5:1 colors specifically for that theme.

Or, make Brett fix it in EditorColors.pkgdef. It's up to you. One benefit of using EditorColors for Blue (Extra Contrast) themeing is that we have tools that verify foregrounds against backgrounds specifically for that theme by analyzing EditorColors. If you do the theming here, it needs to be manually checked more frequently.
